### ISSUE TITLE
Property 'getRawTextContent' does not exist on type 'Pdfparser'.ts(2339)

### ISSUE THREAD

**Issue by zach-betz-hln** [modesty/pdf2json#327]
Thanks for this sweet library.

When using it in a TypeScript project and calling `pdfParser.getRawTextContent()`, I got this error:

```
Property 'getRawTextContent' does not exist on type 'Pdfparser'.ts(2339)
```

In the meantime, I was able to work around it by extending the class:

```ts
import PDFParser from 'pdf2json';

interface PatchedPDFParser extends PDFParser {
  getRawTextContent: () => string;
}

const pdfParser = new PDFParser(undefined, 1) as PatchedPDFParser;

pdfParser.on('pdfParser_dataError', (errorMessage) => {
  console.error(errorMessage);
});

pdfParser.on('pdfParser_dataReady', (_output) => {
  const text = pdfParser.getRawTextContent();
  console.log(text);
});

pdfParser.loadPDF('/path/to/file.pdf');
```

**Comment by modesty** [comment]
please test with [2b47fcd](https://github.com/modesty/pdf2json/commit/2b47fcd336735ee6bf739a6ae9b5ba12271be5dc)

**Comment by zach-betz-hln** [comment]
Hi @modesty - how do I install the package at this particular commit?

**Comment by haahmad** [comment]
seems like a new version with the fix needs to be published to npm

**Comment by zach-betz-hln** [comment]
I've worked around this by creating file `src/@types/pdf2json/index.d.ts` with the below contents:

<details><summary>Contents</summary>

```ts
// Copied and adapted from https://github.com/modesty/pdf2json/blob/master/pdfparser.d.ts

declare module 'pdf2json' {
  declare class PDFParser extends EventEmitter {
    constructor(context?: unknown, needRawText?: number, password?: string);
    parseBuffer(buffer: Buffer, verbosity?: number): void;
    loadPDF(pdfFilePath: string, verbosity?: number): Promise<void>;
    createParserStream(): ParserStream;
    getRawTextContent(): string;
    on<K extends keyof EventMap>(eventName: K, listener: EventMap[K]): this;
  }

  type EventMap = {
    pdfParser_dataError: (errMsg: Record<'parserError', Error>) => void;
    pdfParser_dataReady: (pdfData: Output) => void;
    readable: (meta: Output['Meta']) => void;
    data: (data: Output['Pages'][number] | null) => void;
  };

  declare class ParserStream {
    //TODO
  }

  interface Output {
    Transcoder: string;
    Meta: Record<string, unknown>;
    Pages: Page[];
  }

  interface Page {
    Width: number;
    Height: number;
    HLines: Line[];
    VLines: Line[];
    Fills: Fill[];
    Texts: Text[];
    Fields: Field[];
    Boxsets: Boxset[];
  }

  interface Fill {
    x: number;
    y: number;
    w: number;
    h: number;
    oc?: string;
    clr?: number;
  }

  interface Line {
    x: number;
    y: number;
    w: number;
    l: number;
    oc?: string;
    clr?: number;
  }

  interface Text {
    x: number;
    y: number;
    w: number;
    sw: number;
    A: 'left' | 'center' | 'right';
    R: TextRun[];
    oc?: string;
    clr?: number;
  }

  interface TextRun {
    T: string;
    S: number;
    TS: [number, number, 0 | 1, 0 | 1];
    RA?: number;
  }

  interface Boxset {
    boxes: Box[];
    id: {
      Id: string;
      EN?: number;
    };
  }

  interface Field {
    id: {
      Id: string;
      EN?: number;
    };
    style: number;
    TI: number;
    AM: number;
    TU: string;
    x: number;
    y: number;
    w: number;
    h: number;
    T: {
      Name: 'alpha' | 'link';
      TypeInfo: object;
    };
  }

  interface Box {
    x: number;
    y: number;
    w: number;
    h: number;
    oc?: string;
    clr?: number;
  }

  interface Box {
    id: {
      Id: string;
      EN?: number;
    };
    T: {
      Name: string;
      TypeInfo?: object;
    };
    x: number;
    y: number;
    w: number;
    h: number;
    TI: number;
    AM: number;
    checked?: boolean;
    style: number;
  }

  export default PDFParser;
}
```

</details>

**Comment by StanislavKharchenko** [comment]
Hello.
When the fix will be published?

**Comment by modesty** [comment]
published in 3.1.2
